### PR TITLE
[TIR][Transform] Remove num_unpacked_args from MakePackedAPI

### PR DESCRIPTION
**Pull request by Lunderberg**
Other than a single unit test, there are no usages of this parameter in TVM, and it significantly complicates the logic of `MakePackedAPI`. Similar functionality can be had by using `MakeUnpackedAPI` instead.

### CODE REVIEW

**Comment by Mousius** [comment]
Thanks for doing this @Lunderberg, I've often wondered what this was for 🤔

**Comment by Lunderberg** [comment]
@Mousius No problem!  As far as I could tell from inspection (and maybe @tqchen could verify), it looks like an early-development feature that dates back at least to 2017 (https://github.com/apache/tvm/pull/51).  The usage has mostly been replaced by `MakeUnpackedAPI`, which wasn't present until 2021 (https://github.com/apache/tvm/pull/8023).

File: python/tvm/tir/transform/transform.py [hunk 1]
```diff
@@ -402,7 +402,9 @@ def MakePackedAPI(num_unpacked_params: int = -1):
     fpass : tvm.transform.Pass
         The result pass
     """
-    return _ffi_api.MakePackedAPI(num_unpacked_params)  # type: ignore
+    num_unpacked_params: int = -1
+    assert num_unpacked_params == -1
```
Comment (Mousius) [inline python/tvm/tir/transform/transform.py:406]: I think we can remove this now?

**Review by Mousius — CHANGES_REQUESTED** [review]

File: python/tvm/tir/transform/transform.py (hunk 1 above)
Comment (Lunderberg) [reply python/tvm/tir/transform/transform.py]: Thank you for the catch, and removed!  I was iteratively stripping out `num_unpacked_params` to make sure that each step still ran, and it looks like that one slipped by.

**Pull request merged**